### ISSUE TITLE
error in create_block_mask for H=None

### ISSUE THREAD

**Issue by KonstantinWilleke** [shreyansh26/Attention-Mask-Patterns#1]
Hi there, 

Thanks a lot for sharing this useful repo!

The benchmarks are throwing an error while compiling when creating the masks:
This is the mask code for all the benchmark functions:
```
block_mask = create_block_mask(mask_mod, B=B, H=None, Q_LEN=S, KV_LEN=S, _compile=True)
````
The `H=None` throws an error:

```
Unsupported: TypeError <built-in method arange of type object at 0x7f1bafe5fb80>: arange() received an invalid combination of arguments - got (int, NoneType, device=str), but expected one of:
 * (Number end, *, Tensor out = None, torch.dtype dtype = None, torch.layout layout = None, torch.device device = None, bool pin_memory = False, bool requires_grad = False)
 * (Number start, Number end, *, torch.dtype dtype = None, torch.layout layout = None, torch.device device = None, bool pin_memory = False, bool requires_grad = False)
 * (Number start, Number end, Number step = 1, *, Tensor out = None, torch.dtype dtype = None, torch.layout layout = None, torch.device device = None, bool pin_memory = False, bool requires_grad = False

from user code:
   File "[/usr/local/lib/python3.8/dist-packages/torch/_dynamo/external_utils.py", line 38](http://localhost:10101/usr/local/lib/python3.8/dist-packages/torch/_dynamo/external_utils.py#line=37), in inner
    return fn(*args, **kwargs)
  File "[/usr/local/lib/python3.8/dist-packages/torch/nn/attention/flex_attention.py", line 624](http://localhost:10101/usr/local/lib/python3.8/dist-packages/torch/nn/attention/flex_attention.py#line=623), in _create_block_mask_inner
    mask_tensor = create_mask(mask_mod, B, H, Q_LEN, KV_LEN, device, _compile=True)
  File "[/usr/local/lib/python3.8/dist-packages/torch/nn/attention/flex_attention.py", line 583](http://localhost:10101/usr/local/lib/python3.8/dist-packages/torch/nn/attention/flex_attention.py#line=582), in create_mask
    h = torch.arange(0, H, device=device)
```
setting `H=H` fixes the issue. 
`block_mask = create_block_mask(mask_mod, B=B, H=H, Q_LEN=S, KV_LEN=S, _compile=True)`

**Comment by shreyansh26** [comment]
Interesting, it works perfectly for me with `H=None` as in the [FlexAttention blog](https://pytorch.org/blog/flexattention/) as well, at many places. I'm guessing it is still in beta and hence there may be certain inconsistencies with the Pytorch nightly versions.

Thanks for sharing though!

**Comment by KonstantinWilleke** [comment]
Thanks for your response! Yes it's a bit of a mystery for me, too.
I'm on the exact same pytorch and triton, as specified in your requirements.txt
pytorch `'2.5.0.dev20240910+cu121' and triton `'3.0.0+757b6a61e7'`

But well, torchdynamo is a bit of a mystery.
could also be a cuda/gpu issue. I'm running cuda 12.2 on a L40s. 

In any case, thanks a lot for this useful repo. Cheers!

**Comment by shreyansh26** [comment]
Got it. Yeah it can be a GPU issue. I am also on CUDA 12.2 but ran it on an H100. Well, as long as it is working by providing the `H` value, that's good.

Thanks for your kind words and for trying it out!
